### PR TITLE
Added redux to project

### PR DESCRIPTION
**Pull request by mnaibei**
- Added redux to project
- Created structured folders as required
- Created bookSlice and categorySlice
- Added reducers in both files
- Checked linters

### CODE REVIEW

File: src/redux/store.js
Comment (shella12) [inline src/redux/store.js:1]: - [ ] Great job adding the `store.js` file however you still need to configure the Redux Store (/src/redux/configureStore.js) as mentioned in the[ requirements](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/add_reducers_an_actions.md)


> - Configure the Redux Store (/src/redux/configureStore.js): 
>     - Importing the necessary methods from Redux Toolkit.
>     - Combine both reducers into a root reducer by using configureStore function.
>  

Example snippet:

```js

// ./src/redux/store.js

import { configureStore } from "@reduxjs/toolkit"
import booksReducer from "./books/booksSlice"
import categoriesReducer from "./categories/categoriesSlice"

const store = configureStore({
  reducer: { books: booksReducer, categories: categoriesReducer },
})

export default store

```

**Review by shella12 — CHANGES_REQUESTED** [review]
Hi @mnaibei, 

Good job so far, indeed you have spent time on this project!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- Used `ducks pattern` ✅ 
- Added book's actions and reducer ✅   
- Added categories actions and reducer ✅ 
- Installed `@reduxjs/toolkit` & `react-redux` ✅ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/store.js
Comment (mnaibei) [reply src/redux/store.js:1]: @shella12 Thanks for that. I have added it to my file

**Review by emmiiorji — APPROVED** [review]
Hi @mnaibei, 

# Status: Approved ✅

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

<div align='center'>

</div>

## Good Highlights

- Redux store has been set up as requested

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**